classifiers/text_analysis/emotionality_detection/code_snippet_common.md

```diff
@@ -1,26 +1,38 @@
 ```python
-from LeXmo import LeXmo
 
-def emotionality_detection(text:str) -> str:
-    """
-    @param text: text to check
-    @return: either 'anger', 'fear', 'anticipation', 'trust', 'surprise', 'sadness', 'joy' or 'disgust' depending on the score
-    """
-    emo = LeXmo.LeXmo(text)
-    del emo["text"]
-    del emo["positive"]
-    del emo["negative"]
-    emo = max(emo, key=emo.get)
-    return emo
-    
+import requests
+import spacy
+
+def emotionality_detection(text, api_key):
+      headers = {"Authorization": f"Bearer {api_key}"}
+      data = {"inputs": text}
+      try: 
+            response = requests.post("https://api-inference.huggingface.co/models/j-hartmann/emotion-english-distilroberta-base", headers=headers, json=data)
+            response_json = response.json()
+            ner_positions = []
```

Comment:
This is supposed to be a classification brick and not an extraction one. This code tries to extract entities from a text and yield them, which is not how this model works. The output of the model returns a list containing a dict like this: [[{'label': 'disgust', 'score': 0.8062547445297241}, {'label': 'sadness', 'score': 0.107185959815979}, {'label': 'anger', 'score': 0.06451956927776337}, {'label': 'fear', 'score': 0.014504954218864441}, {'label': 'neutral', 'score': 0.0030063888989388943}, {'label': 'joy', 'score': 0.0026550842449069023}, {'label': 'surprise', 'score': 0.0018732782918959856}]]

Doing NER won't work.